### ISSUE TITLE
False Negative: Unused private property not reported in a file if used in another class

### ISSUE THREAD

**Issue by xouabita** [detekt/detekt#6702]
## Expected Behavior

```kt
class A(private val foo: Any) {
    fun myFoo() = foo
}
class B(private val foo: Any) // <-- foo should be reported here
```

## Observed Behavior

It is not reprorted because the `unusedParameters` list is per file: https://github.com/detekt/detekt/blob/9b2e1eace3d50a8e19f3ce6b1b43bce0696bbcbd/detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnusedParameter.kt#L68

So `foo` is marked as used because of the usage in class `A`, however it is unused in class `B`

## Steps to Reproduce

I have created a PR to reproduce: https://github.com/detekt/detekt/pull/6701
